### DISCUSSION TITLE
Dispatching `undefined` as parameter doesn't work

### DISCUSSION THREAD

**Discussion by voluntadpear** [alpinejs/alpine#2106] [7. Bugs (archived)]
I'm using `$dispatch("input", value)` as a way to then bind my Alpine.js component with `x-model`. However, I've noticed that when `value` is `undefined` the change is not correctly propagated.

This is because of https://github.com/alpinejs/alpine/blob/b46c41fa240cd8af2dcaa29fb60fb1db0389c95a/packages/alpinejs/src/utils/dispatch.js#L2, it is explicitly falling back to an empty object if `details` is not specified.

I think trying to do something like `$dispatch("input", undefined)` makes sense for certain use cases like mine, where I want to explicitly cleared the current value of my custom component, and people using `x-model` on my custom component expecting the correct behavior.

**Reproduction link:** https://stackblitz.com/edit/web-platform-whsb44?file=index.html

**Workaround using custom event:** https://stackblitz.com/edit/web-platform-tr9adr?file=index.html

**Comment by SimoTod** [comment]
I don't think undefined is conceptually right. If you want to reset it, pass the empty string (your original value), even null would work but that's not exactly a reset. Undefined is a bit of a weird value in js, if you pass it to a function, it's like not passing anything.

**Reply by voluntadpear** [reply]
Thanks for the answer! I think it's still unexpected that passing `undefined` as the value results in receiving an empty object in the other end.

I think it should be documented at least, because otherwise, it's a bit counterintuitive. I've used Vue.js in the past and `v-model` works just fine with emitting (their equivalent to dispatching) `undefined`: https://stackblitz.com/edit/vue-ntypad?file=src/App.vue.

**Reply by SimoTod** [reply]
In my opinion, a developer should never pass undefined because that means "i don't have the variable, use your default value" and because you are not supposed to know about the internal implementation that an author should be free to change at any time, you can't predict the outcome.

The documentation is open if you want to send a PR anyway (you should probably document what happen if you pass undefined on any function at this point, though)
